drivers/pds/pdsbackup/backuputils.go

```diff
@@ -0,0 +1,217 @@
+		return err
+	}
+	time.Sleep(bkpMaxtimeInterval)
+	_, err = backupClient.components.BackupCredential.DeleteBackupCredential(credId)
+	if err != nil {
+		log.Errorf("Failed to delete AWS S3 backup credentials, Err: %v ", err)
+		return err
+	}
+	log.Info("Delete S3 bucket from AWS cloud.")
+	err = backupClient.awsStorageClient.deleteBucket(bucketName)
+	if err != nil {
+		log.Errorf("Failed to delete S3 bucket %s, Err: %v ", bucketName, err)
+		return err
+	}
+	return nil
+}
+
+// DeleteAzureBackupCredsAndTarget delete backup creds,bucket and target.
+func (backupClient *BackupClient) DeleteAzureBackupCredsAndTarget(backupTargetId string) error {
+	log.Info("Removing Azure backup creadentials and target from PDS.")
+	backupTarget, _ := backupClient.components.BackupTarget.GetBackupTarget(backupTargetId)
+	credId := backupTarget.GetBackupCredentialsId()
+	log.Infof("Deleting backup target {Name: %v} to PDS.", backupTarget.GetName())
+	_, err := backupClient.components.BackupTarget.DeleteBackupTarget(backupTargetId)
+	backupTarget.GetBackupCredentialsId()
+	if err != nil {
+		log.Errorf("Failed to delete Azure backup target, Err: %v ", err)
+		return err
+	}
+	time.Sleep(bkpMaxtimeInterval)
+	_, err = backupClient.components.BackupCredential.DeleteBackupCredential(credId)
+	if err != nil {
+		log.Errorf("Failed to delete Azure backup credentials, Err: %v ", err)
+		return err
+	}
+
+	log.Info("Delete Azure blob storage")
+	err = backupClient.azureStorageClient.deleteBucket(bucketName)
+	if err != nil {
+		log.Errorf("Failed to delete  azure blob %s, Err: %v ", bucketName, err)
+		return err
+	}
+	return nil
+}
+
+// DeleteGoogleBackupCredsAndTarget delete backup creds,bucket and target.
+func (backupClient *BackupClient) DeleteGoogleBackupCredsAndTarget(backupTargetId string) error {
+	log.Info("Removing Google backup creadentials and target from PDS.")
+	backupTarget, _ := backupClient.components.BackupTarget.GetBackupTarget(backupTargetId)
+	credId := backupTarget.GetBackupCredentialsId()
+	log.Infof("Deleting backup target {Name: %v} to PDS.", backupTarget.GetName())
+	_, err := backupClient.components.BackupTarget.DeleteBackupTarget(backupTargetId)
+	if err != nil {
+		log.Errorf("Failed to delete Google backup target, Err: %v ", err)
+		return err
+	}
+	time.Sleep(bkpMaxtimeInterval)
+	_, err = backupClient.components.BackupCredential.DeleteBackupCredential(credId)
+	if err != nil {
+		log.Errorf("Failed to delete Google backup credentials, Err: %v ", err)
+		return err
+	}
+	log.Info("Deleting google storage.")
+	err = backupClient.gcpStorageClient.deleteBucket(bucketName)
+	if err != nil {
+		log.Errorf("Failed to delete bucket %s, Err: %v ", bucketName, err)
+		return err
+	}
+	return nil
+}
+
+// InitializePdsBackup to create backup creds/targets.
+func InitializePdsBackup() (*BackupClient, error) {
+
+	envVars := pdsutils.BackupEnvVariables()
+	apiConf := pds.NewConfiguration()
+	endpointURL, err := url.Parse(envVars.PDSControlPlaneURL)
```

Comment:
see if we can use the control plane URL from the PDS_PARAMS file

Reply:
done